klient/conf/config.go

```diff
@@ -33,14 +34,34 @@ var DefaultClusterContext = ""
 // New returns Kubernetes configuration value of type *rest.Config.
 // filename is kubeconfig file
 func New(fileName string) (*rest.Config, error) {
-	// if filename is not provided assume in-cluster-config
+	var resolvedKubeConfigFile string
+	kubeContext := ResolveClusterContext()
+
+	// resolve the kubeconfig file
 	if fileName == "" {
-		return rest.InClusterConfig()
+		resolvedKubeConfigFile = ResolveKubeConfigFile()
+		if fileName != "" {
+			resolvedKubeConfigFile = fileName
+		}
+	}
```

Comment:
Sorry, I missed this one 🤦 

This can't be happening right ? Your first check is `fileName == ""` and inside the lop you have a `fileName != ""` without first setting the `fileName` to a new values.

Reply:
@maruina  I think I know why. Check the above one. 

```golang
// resolve the kubeconfig file
	resolvedKubeConfigFile = fileName
	if fileName == "" {
		resolvedKubeConfigFile = ResolveKubeConfigFile()
	}
```

Should be like above.

Reply:
Yes, of course 🤦 
Tests are passing locally now.